include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,151 @@
+//======================================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/detail/skeleton.hpp>
+#include <eve/detail/function/conditional.hpp>
+
+namespace eve
+{
+  namespace detail
+  {
+    inline constexpr struct { EVE_FORCEINLINE auto operator()(auto, auto x) const { return x; } } return_2nd = {};
+  }
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct callable : decorated_with<OptionsValues, Options...>
+  {
+    using base = decorated_with<OptionsValues, Options...>;
+
+    template<typename T>
+    constexpr auto operator[](T t) const
+    requires( requires(base const& b) { b[t];} )
+    {
+      auto new_traits = base::operator[](t);
+      return  Func<decltype(new_traits)>{new_traits};
+    }
+
+    template<typename T> void operator[](T t) const =delete;
+
+    template<typename... Args>
+    constexpr auto behavior(auto arch, Args&&... args) const
+    {
+      return Func<OptionsValues>::deferred_call(arch, EVE_FWD(args)...);
+    }
+
+    protected:
+    constexpr Func<OptionsValues> const& derived() const { return static_cast<Func<OptionsValues>>(*this); }
+  };
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  {
+    template<callable_options O, typename T, typename... Ts>
+    constexpr auto behavior(auto arch, O const& opts, T x0,  Ts const&... xs) const
+    {
+      using func_t                =  Func<OptionsValues>;
+      constexpr bool is_supported = requires{ func_t::deferred_call(arch, opts, x0, xs...); };
+      constexpr bool any_simd     = (simd_value<T> || ... || simd_value<Ts>);
+
+      // Are we dealing with a no-condition call ?
+      if constexpr( option_type_is<condition_key, O, ignore_none_> )
```

Comment:
* Ignore all support?

Do we have a uniform logic on how we handle ignores in element wise?

Reply:
I think this is a bit lacking indeed